docs/index.asciidoc

```diff
@@ -0,0 +1,50 @@
+:plugin: example
+:type: codec
+
+///////////////////////////////////////////
+START - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+:version: %VERSION%
+:release_date: %RELEASE_DATE%
+:changelog_url: %CHANGELOG_URL%
+:include_path: ../../../../logstash/docs/include
+///////////////////////////////////////////
+END - GENERATED VARIABLES, DO NOT EDIT!
+///////////////////////////////////////////
+
+[id="plugins-{type}s-{plugin}"]
+
+=== Example codec plugin
+
+include::{include_path}/plugin_header.asciidoc[]
+
+==== Description
+
+ADD DESCRIPTION HERE
+
+[id="plugins-{type}s-{plugin}-options"]
+==== Example Codec Configuration Options
+
+[cols="<,<,<",options="header",]
+|=======================================================================
+|Setting |Input type|Required
+| <<plugins-{type}s-{plugin}-setting_name>> |{logstash-ref}/configuration-file-structure.html#boolean[boolean]|No
+| <<plugins-{type}s-{plugin}-another_setting_name>> |{logstash-ref}/configuration-file-structure.html#hash[hash]|No
+|=======================================================================
+
+[id="plugins-{type}s-{plugin}-setting_name"]
+===== `setting_name` 
+
+  * Value type is {logstash-ref}/configuration-file-structure.html#boolean[boolean]
+  * Default value is `true`
+
+ADD DESCRIPTION HERE.
+
+[id="plugins-{type}s-{plugin}-another_setting_name"]
+===== `another_setting_name` 
+
+  * Value type is <<hash,hash>>
+  * Default value is `{}`
+
+ADD DESCRIPTION HERE.
+
```

Comment:
Looks like there are no objections to adding a sample gemspec, so let's add one.  
We should following the new changelog guidelines in  logstash/plugin_changelog_guidelines.md. 
@dedemorton, would you like for me to take this?  Let's discuss!

Reply:
@karenzone Yes please take this.

Reply:
I've done the work, but don't have permissions to push to your branch.  If you want to give me permissions, I'll push.  If you'd rather merge your PR and have me follow up with a separate PR, that works, too.  I'm good with whatever is easiest for you.

Reply:
@karenzone You can just create a new PR with your changes (which I assume you've made on top of the changes from my PR) and then close my PR.

Reply:
TIL. Will do. Thanks!